main.go

```diff
@@ -0,0 +1,453 @@
+	glog.V(10).Infof("      Total Users in Organization %d", len(allUsers))
+
+	// Launch goroutines for each user and find which projects and organizations they have access to
+	for _, u := range allUsers {
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			// Select a random service account to use per user and distribute quota consumption
+			impersonateAccount, svcKeyCred, err := getRandomServiceAccount(serviceAccounts, svcAccountKeys)
+			if err != nil {
+				glog.Errorf("Error finding valid serviceAccount for user %s %v", u.PrimaryEmail, err)
+				return
+			}
+
+			// Get a resource manager API client that acts as the current user
+			crmService, err := getResourceManagerClient(ctx, impersonateAccount, svcKeyCred, u.PrimaryEmail)
+			if err != nil {
+				glog.Errorf("Error getting ResourceManager client for user %s %v", u.PrimaryEmail, err)
+				return
+			}
+
+			// Find all the organizations and projects the user has access to
+
+			// The organizationFilter can be used to limit the asset api query.
+			// In almost all cases, this should be empty since we want to see all organizations visible to the user.
+			// see https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v3#OrganizationsSearchCall.Query
+
+			// Projects.Search() accepts a Filter parameter which will return a subset
+			// of projects that match the specifications.
+			// By default, if the Filter value is not set and all projects will be returned.
+			// However, the code below leaves the parameter
+			// explicitly defined in the event you want to query and evaluate on a
+			// subset of projects. projectFilter="lifecycleState=ACTIVE"
+			// See:
+			// https://cloud.google.com/resource-manager/reference/rest/v3/organizations/search
+			// see https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v3#ProjectsSearchCall.Query
+			organizationFilter := ""
+			projectFilter := ""
+			userOrganizations, err := getOrganizations(ctx, limiter, organizationFilter, *crmService, *u)
+			if err != nil {
+				glog.Errorf("Error finding Organizations for user %s %v", u.PrimaryEmail, err)
+				// don't return, try to get the projects at least.
+			}
+			// We are only interested in the projectNumber (which is in the name field) and ID so restrict the response to just those fields
+			// see https://developers.google.com/gdata/docs/2.0/basics#PartialResponse
+			fields := "nextPageToken,projects(name,projectId)"
+			userProjects, err := getProjects(ctx, limiter, projectFilter, fields, *crmService, *u)
+			if err != nil {
+				glog.Errorf("Error finding Projects for user %s %v", u.PrimaryEmail, err)
+				return
+			}
+
+			// Find the projects and organizations that do NOT exist in the list of projects under the subject organization.
+			// The projects the user has access to that is not included in []allProjects should be outside the subject organization.
+			for _, o := range userOrganizations {
```

Comment:
Any issue with interleaving log statements about different users? Might want a mutex. But, probably not a big deal.

Reply:
nope, no issue with interleaving stuff,  its pretty easy for a user to sort in order from the output